core/coreapi/coreapi.go

```diff
@@ -225,12 +226,20 @@ func (api *CoreAPI) WithOptions(opts ...options.ApiOption) (coreiface.CoreAPI, e
 			return nil, fmt.Errorf("cannot specify negative resolve cache size")
 		}
 
-		subAPI.routing = offlineroute.NewOfflineRouter(subAPI.repo.Datastore(), subAPI.recordValidator)
-
-		subAPI.namesys, err = namesys.NewNameSystem(subAPI.routing,
+		nsOptions := []namesys.Option{
 			namesys.WithDatastore(subAPI.repo.Datastore()),
 			namesys.WithDNSResolver(subAPI.dnsResolver),
-			namesys.WithCache(cs))
+			namesys.WithCache(cs),
+		}
+
+		if !cfg.Ipns.MaxCacheTTL.IsDefault() {
+			// Default value won't be used since we check for it. There is no default.
+			nsOptions = append(nsOptions, namesys.WithMaxCacheTTL(cfg.Ipns.MaxCacheTTL.WithDefault(time.Second)))
+		}
```

Comment:
@hacdias  fysa I simplified this a bit (https://github.com/ipfs/kubo/pull/10321/commits/5187a45ad5824b28d75e2ccbc81a24b7ab293824)  for the sake of maintenance. Defined `config.DefaultIpnsMaxCacheTTL` as `math.MaxInt64`, and used that as idiomatic default, this way we don't have any additional boilerplate around default value.

I also made it possible to set this to 0, but with proper warning in docs.